hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -563,6 +622,7 @@ public AbfsRestOperation getPathStatus(final String path, final boolean includeP
   public AbfsRestOperation read(final String path, final long position, final byte[] buffer, final int bufferOffset,
                                 final int bufferLength, final String eTag, String cachedSasToken) throws AzureBlobFileSystemException {
     final List<AbfsHttpHeader> requestHeaders = createDefaultHeaders();
+    addCustomerProvidedKeyHeaders(requestHeaders);
```

Comment:
For the purposes of debugging, add a log line if request fails and the sha256 of the key used in request and the server returned sha256 in response are different.

Reply:
Server doesn't send back the actual sha of the encryption key with which the data is encrypted if the encryption key sent is wrong